### PR TITLE
Feat/add metrics for rebalancing

### PR DESCRIPTION
**Pull request by abhidtu2014**
[Task](https://github.com/XLabs/wallet-monitor/issues/91)

## Description
Rebalancing metrics that needs to be added
- [ ] expenditure by chain (usd)
- [x] wallet-pool available wallets
- [x] expenditure by chain
- [x] rebalancing count
- [x] rebalancing errors

## Testing

Tested metrics on grafana locally using ganache setup (partially)

Screenshot 2023-09-22 at 4 58 52 PM

Screenshot 2023-09-25 at 1 11 47 PM

### CODE REVIEW

File: src/prometheus-exporter.ts [hunk 1]
```diff
@@ -109,10 +143,14 @@ export class PrometheusExporter {
     this.prometheusPath = path || '/metrics';
 
     this.balancesGauge = createBalancesGauge(this.registry, 'wallet_monitor_balance');
+    this.activeWalletsGauge = createActiveWalletsGauge(this.registry, 'wallet_monitor_active_wallets');
+    this.walletsLockPeriodGauge = createWalletsLockPeriodGauge(this.registry, 'wallet_monitor_wallets_lock_period')
     this.rebalanceCounter = createRebalanceCounter(this.registry, 'wallet_monitor_rebalance');
     this.rebalanceExpenditureCounter = createRebalanceExpenditureCounter(this.registry, 'wallet_monitor_rebalance_expenditure');
     this.rebalanceInstructionsCounter = createRebalanceInstructionsCounter(this.registry, 'wallet_monitor_rebalance_instruction');
     this.rebalanceTransactionsCounter = createRebalanceTransactionsCounter(this.registry, 'wallet_monitor_rebalance_transaction');
+    this.rebalanceSuccessCounter = createCounter(this.registry, "wallet_monitor_rebalance_success_total", "Total number of successful rebalances", ['chain_name', 'strategy']);
```
Comment (mat1asm) [inline src/prometheus-exporter.ts:152]: It appears we already have a rebalances counter, and now we would add failure and success counters. Maybe we can have only one? Or if we really need to have a separate one for the number of times rebalances start, we could probably merge the new ones into `finishedRebalanceCounter`, adding a status label.

**Review by mat1asm — APPROVED** [review]
Added a small comment to review, still LGTM

File: src/prometheus-exporter.ts (hunk 1 above)
Comment (abhidtu2014) [reply src/prometheus-exporter.ts:152]: Agree, we can reduce it into one counter with status as `success` and `failed` and with both statuses, we can retrieve total rebalances. Thanks for pointing out. I will update the code.

Comment (abhidtu2014) [reply src/prometheus-exporter.ts]: updated

**Comment by abhidtu2014** [comment]
As discussed with @solanoepalacio :
- we need to remove `transactionsCounter`
- we need to update `instructionsCounter` with label `success` and `failed`


PS: Please correct me if I am wrong.

File: examples/README.md
Comment (solanoepalacio) [inline examples/README.md:1]: :black_heart:

File: examples/rebalancing.ts [hunk 2]
```diff
@@ -49,10 +99,39 @@ const allChainWallets: WalletManagerConfig = {
...
+};
 
-}
+export const manager = buildWalletManager({ config: allChainWallets, options });
 
 
-export const manager = buildWalletManager({config: allChainWallets, options})
+for (let i = 0; i < 10; i++) {
+  // perform an action with any wallet available in the pool:
+  manager.withWallet("ethereum", async wallet => {
```
Comment (solanoepalacio) [inline examples/rebalancing.ts:111]: :black_heart:

**Review by solanoepalacio — APPROVED** [review]

**Pull request merged**